pkg/serviceprovider/access_token_mapper.go

```diff
@@ -30,6 +30,11 @@ import (
 
 // Key for token using in Opaque Secret
 const tokenKey = "token"
+const (
+	dockerConfigJsonTypeAnnotationKey     = "spi.appstudio.redhat.com/config-json-type"
+	dockerConfigJsonExplicitAnnotationKey = "spi.appstudio.redhat.com/config-json-auth-key"
+	dockerConfigJsonTypeKubernetes        = "kubernetes"
```

Comment:
To be maximally explicit, I would have 3 `dockerConfigJSonType`s:
* `docker` - for setting the host as the auth key, as docker does,
* `kubernetes` - setting the full provider URL as the auth key,
* `explicit` - for setting the auth key using the value in the `spi.appstudio.redhat.com/config-json-auth-key`.

That way, the configuration is explicit and we don't need to talk about the precedence of the values, etc.

Reply:
Good idea.